barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -346,11 +358,13 @@ template <typename Curve_> class IPA {
             }
             // TODO(https://github.com/AztecProtocol/barretenberg/issues/1140): Use batch_invert.
             round_challenges_inv[i] = round_challenges[i].invert();
+            if (i < log_poly_length) {
```

Comment:
for native verifier, we only need to look at the log_poly_length elements that matter